### PR TITLE
refactor: split Passage enter/exit from Orders

### PR DESCRIPTION
**Pull request by anna-carroll**
current PR TODO 
- [ ] prettier way of dealing with `Fulfiller`/`Initiator` inheritance
- [ ] fix however this probably breaks CI (e.g. 2 contracts now instead of 1)

future TODO
- [ ] token factory for mint/burn ERC20s on the RU 
- [ ] `transact` flow (also affects ERC20s path)
- [ ] `cross` flow?

### CODE REVIEW

**Comment by anna-carroll** [comment]
i do think this is conceptually cleaner.

- `Passage` — enter/exit — manages mint/burn Ether and tokens 
   - custodies TVL of the rollup on host chain.
      - only withdrawalAdmin multisig can take funds out on the host.
   - holds ERC20 mint/burn permissions on rollup.
      - accessible via `transact` flow via `hostPassage` address.
   - Node must watch this contract to mint/burn native Ether.
   - (soon) Node must watch this contract for `transact` flow.

- `Market` — swap/fill/sweep — manages intent-based Orders
   - doesn’t custody any funds or permissions.
   - Node must watch this contract to check for intent fulfillment.

**Review by prestwich — COMMENTED** [review]
directionally I approve. seems there are some TODOs

**Comment by anna-carroll** [comment]
note: ERC20 enters/exits are not specced yet, so i'm removing those functions from this PR for now. 

if we decide not to implement them, it would make more sense to remove the `token` field from the `Enter` event (as it would always be `address(0)` - native ether). 

let's decide on ERC20 plan then ossify that event interface.

File: src/Passage.sol [hunk 1]
```diff
@@ -104,83 +69,19 @@ contract Passage is BasePassage {
...
-            balance = IERC20(token).balanceOf(address(this));
-            IERC20(token).transfer(recipient, balance);
-        }
-        emit Sweep(token, recipient, balance);
+/// @notice A contract deployed to the Rollup that allows users to remove tokens from the Rollup TVL back to the Host.
+contract RollupPassage {
+    /// @notice Emitted when tokens exit the rollup.
+    /// @param token - The address of the token exiting the rollup.
+    /// @param recipient - The desired recipient of the token on the host chain.
+    /// @param amount - The amount of the token entering the rollup.
```
Comment (prestwich) [inline src/Passage.sol:77]: typo: exiting

File: src/Passage.sol [hunk 2]
```diff
@@ -104,83 +69,19 @@ contract Passage is BasePassage {
...
+    event Exit(address indexed token, address indexed recipient, uint256 amount);
+
+    /// @notice Allows native Ether to exit the rollup.
+    /// @dev Rollup node will burn the msg.value.
+    /// @param recipient - The desired recipient of the Ether on the host chain.
+    /// @custom:emits Exit indicating the amount of Ether to burn on the rollup & the recipient on the host chain.
+    function exit(address recipient) public payable {
```
Comment (prestwich) [inline src/Passage.sol:84]: probably out of scope for this PR as it's unspecified new behavior in a refactor

**Review by prestwich — APPROVED** [review]
i would delete the exit code, as we haven't specified this behavior, and we don't know if the code is appropriate for it

File: src/Passage.sol (hunk 1 above)
Comment (anna-carroll) [reply src/Passage.sol:77]: Screenshot 2024-06-10 at 3 33 33 PM
🤔

Comment (prestwich) [reply src/Passage.sol:77]: L77

Comment (anna-carroll) [reply src/Passage.sol:77]: ahhh i see

**Pull request merged**